lib/dgram.js

```diff
@@ -324,15 +324,20 @@ Socket.prototype.sendto = function(buffer,
 function sliceBuffer(buffer, offset, length) {
   if (typeof buffer === 'string') {
     buffer = Buffer.from(buffer);
-  } else if (!isUint8Array(buffer)) {
+  } else if (!isArrayBufferView(buffer)) {
     throw new ERR_INVALID_ARG_TYPE('buffer',
-                                   ['Buffer', 'Uint8Array', 'string'], buffer);
+                                   ['Buffer',
+                                    'TypedArray',
+                                    'DataView',
+                                    'string'],
+                                   buffer);
   }
 
   offset = offset >>> 0;
   length = length >>> 0;
 
-  return buffer.slice(offset, offset + length);
+  return Buffer.from(buffer.buffer, buffer.byteOffset, buffer.byteLength)
+          .slice(offset, offset + length);
```

Comment:
I think you can get rid of the extra `.slice()` call by adding the `offset` and `length` in the `Buffer.from()` call?

Reply:
if `length` were greater than `buffer.byteLength`, using `.slice()` separately would default to limiting number of bytes exposed to `buffer.byteLength`.

if `length` were smaller than `buffer.byteLength`, using `.slice()` separately would further reduce the number of bytes exposed.

would something like this achieve the same behaviour?
```
const maxBytesExposed = (length > buffer.byteLength) ? buffer.byteLength : buffer.byteLength - length;

return Buffer.from(buffer.buffer, buffer.byteOffset + offset, maxBytesExposed);
```

Reply:
@SirR4T I think if the offsets/length for `Buffer.from()` are out of bounds, the extra data just gets discarded. I think this should work:

```js
Buffer.from(buffer.buffer, buffer.byteOffset + offset, length);
```

?

Reply:
done.

Reply:
@addaleax this behaviour wasn't obvious to me, from the [`Buffer.from(arrayBuffer, offset, length)` docs](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_arraybuffer_byteoffset_length). Should we add it to the docs? or is it already somewhere and i missed it?